### PR TITLE
Refactor All DI modules inside a Modules.kt file

### LINKED ISSUE #290: Refactor All DI modules inside a Modules.kt file.
**Describe the feature you'd like**

Refactor the code of the below 4 files inside a Modules.kt file. 

Screenshot 2020-01-07 at 5 22 26 PM
 
**Would you like to work on the issue?**
No

### PR DESCRIPTION
**Pull request by Darshpreet2000**
Fixes #290 

Refactored all the files in di package in a single file Modules.kt

## Screenshots
No changes in screen

## Open questions and concerns
Please review my changes it also includes unit test of splash screen in earlier PR.

### CODE REVIEW

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/di/Modules.kt [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+
+/**
+ * Use the [provideCommonsRetrofit] to provide a Retrofit with WIKI_MEDIA_COMMONS instance
+ * @return  Retrofit*/
+fun provideCommonsRetrofit(): Retrofit =
+        Retrofit.Builder()
+                .baseUrl(Urls.WIKIMEDIA_COMMONS_BASE_URL)
+                .client(provideClient())
+                .addConverterFactory(GsonConverterFactory.create(providerGson()))
+                .build()
```
Comment (ujjwalagrawal17) [inline app/src/main/java/org/wikiedufoundation/wikiedudashboard/di/Modules.kt]: If possible let's have it in another file as it is not related to DI.

Comment (Darshpreet2000) [reply app/src/main/java/org/wikiedufoundation/wikiedudashboard/di/Modules.kt]: Okay ,I will place this in its original file

Comment (Darshpreet2000) [reply app/src/main/java/org/wikiedufoundation/wikiedudashboard/di/Modules.kt]: I am adding this back in apiModule.kt

Comment (Darshpreet2000) [reply app/src/main/java/org/wikiedufoundation/wikiedudashboard/di/Modules.kt]: In which file and which package?

File: app/src/test/java/org/wikiedufoundation/wikiedudashboard/SplashScreenTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import org.wikiedufoundation.wikiedudashboard.ui.splash.SplashActivity;
+import org.wikiedufoundation.wikiedudashboard.ui.welcome.onboarding.WelcomeHostActivity;
+
+import static org.junit.Assert.assertEquals;
+import static org.robolectric.Shadows.shadowOf;
+@RunWith(RobolectricTestRunner.class)
+public class SplashScreenTest {
+   private Intent expectedIntent;
+    @Test
+    public void splashscreen() throws Exception
```
Comment (ujjwalagrawal17) [inline app/src/test/java/org/wikiedufoundation/wikiedudashboard/SplashScreenTest.java]: splashscreen -> splashScreenUITest

**Comment by Darshpreet2000** [comment]
Should I change file and function name here?

**Comment by ujjwalagrawal17** [comment]
Function Name.

**Comment by Darshpreet2000** [comment]
Please merge this PR

**Comment by ujjwalagrawal17** [comment]
Thanks @Darshpreet2000 .
